### PR TITLE
use rwlock replace lock

### PR DESCRIPTION
**Pull request by knightXun**
A little change: use RWLock replace Lock, hope to improve some performance

### CODE REVIEW

File: pkg/mcp/snapshot/inmemory.go [hunk 1]
```diff
@@ -24,11 +24,14 @@ import (
...
 	"github.com/gogo/protobuf/types"
 
+	"sync"
+
 	mcp "istio.io/api/mcp/v1alpha1"
 )
 
 // InMemory Snapshot implementation
 type InMemory struct {
+	mutex     sync.RWMutex
```
Comment (ozevren) [inline pkg/mcp/snapshot/inmemory.go]: There shouldn't be a need for mutexes in the snapshot code. It uses builder/immutable data pattern.

Comment (knightXun) [reply pkg/mcp/snapshot/inmemory.go]: ok

**Comment by knightXun** [comment]
ping @ayj

**Comment by sdake** [comment]
@knightXun I've rerun the circleci required jobs.  If that fails agian, can you please  rebase your PR?

Cheers
-steve

**Comment by knightXun** [comment]
@sdake ok

**Comment by knightXun** [comment]
@sdake the failure of link test seems weird

**Comment by knightXun** [comment]
@sdake cheers, except racetest, I think my pr can be merged.

**Pull request merged**